solidity/contracts/OracleAggregator.sol

```diff
@@ -87,6 +88,12 @@ contract OracleAggregator is AccessControl, SimpleOracle, IOracleAggregator {
     return _availableOracles;
   }
 
+  /// @inheritdoc IOracleAggregator
+  function previewAddOrModifySupportForPair(address _tokenA, address _tokenB) external view returns (ITokenPriceOracle) {
+    OracleAssignment memory _assignment = assignedOracle(_tokenA, _tokenB);
+    return _canModifySupportForPair(_tokenA, _tokenB, _assignment) ? _findFirstOracleThatCanSupportPair(_tokenA, _tokenB) : _assignment.oracle;
+  }
```

Comment:
Thinking out loud, and scared of your answer that might mean 300 PRs... Should this be in `ITokenPriceOracle`? 👀

Reply:
But this only makes sense in the aggregator. I mean, the Uniswap oracle doesn't assign an oracle. Makes sense?